pkg/config/config.go

```diff
@@ -93,9 +95,9 @@ type Webhook struct {
 }
 
 type Install struct {
-	Automatic bool               `json:"automatic,omitempty"`
-	Mode      string             `json:"mode,omitempty"`
-	Networks  map[string]Network `json:"networks,omitempty"`
+	Automatic           bool    `json:"automatic,omitempty"`
+	Mode                string  `json:"mode,omitempty"`
+	ManagementInterface Network `json:"managementInterface,omitempty"`
```

Comment:
How about naming it `ManagementNetwork`? Since it's a Network type and inside it, there is already a field `Interfaces`.

Reply:
I'm not sure
ManagementNetwork might conflict with the term of the network of Canal.

Reply:
It's a field name in Harvester configuration; why does it conflict with Canal things?

And do we need to remove the original `Networks` map? We can keep the `Networks` map but only allow the `harvester-mgmt` key here. This provides backward compatibility. For users who don't have other networks configuration, they don't need to alter their existing configuration files.

Reply:
> Centralized Gateway[¶](https://kubeovn.github.io/docs/v1.10.x/en/guide/subnet/#centralized-gateway)

It will confuse us, we will not sure if we are talking about Pod's Management Network or Management Interface (node's network)

> And do we need to remove the original Networks map? We can keep the Networks map but only allow the harvester-mgmt key here. This provides backward compatibility. For users who don't have other networks configuration, they don't need to alter their existing configuration files.

I will prefer to remove `harvester-mgmt` term in future version.
Make sure it will not confuse anyone.

Reply:
It will not because [we have the document](https://docs.harvesterhci.io/v1.0/install/harvester-configuration/#installnetworks). We don't present the field to the end-user in the installer UI either. If it confuses anyone, we can add a better document or UI terms.

Harvester is already GA for almost one year, and we already have a user base. So keeping more compatibility here would be appreciated.

Reply:
> Harvester is already GA for almost one year, and we already have a user base. So keeping more compatibility here would be appreciated.

I didn't really get it about the naming.
If we remove the `harvester-mgmt` and it will never be showed in Linux system.
And if we keep the naming in installer config, won't it confuse the users?

It will only affect in fresh installation but not upgrading.
Nodes from previous version will not be affected at all.

It will be a little bit weird in docs.
we ask users to write config with `harvester-mgmt`, but this keyword doesn't mean anything.

Reply:
The key here is compatibility and not bringing too much impact to existing users.
I understand we do try our best to migrate existing configurations in the upgrade, but we also need to take care of existing users who have already an infrastructure to use Harvester. For example:
- Customers/users who already have their infrastructure code to spawn Harvester on demand.
- Our QA: they do have infrastructure codes to deploy Harvester repeatedly.
- Our lab maintainer.
- Equinix Metal users.
- Any developers who re-deploy Harvester daily.

Even if we keep the original `Networks`, we still need to check if there are additional configured network keys because they are no longer supported. What I really suggest is to minimize impact. Not arguing the naming.

Reply:
I will suggest, if we put so many changes in this verison, we should consider to take this chance to migrate all we should.
From my side, I will not consider to leave some legacy entries if we don't need it and that is not for a patch build.

and also you mentioned the additional network keys are not supported anymore, that means some users still need to change the script or be noticed some changes to make sure the scripts will fit the new installer config format.

I also minimize impact for the future change to make sure there is no other will confuse with that entry and code.

But you reminded me, that we should ask users to add `scheme_version` to their config.yaml in the future if we consider to have scheme version for future migration plan.

Reply:
both design has pros and cons, if upgrade capability and customer support is our main concerns we may just ask feedbacks internally, and be aware of need to adding those changes to the ipxe repo and existing infra code.

Reply:
The user can know configuration changes:
![Screen Shot 2022-09-20 at 2 49 28 PM](https://user-images.githubusercontent.com/1691518/191187106-1802ccb6-665a-4a2f-b0c4-7cbb910e5196.png)

For the scheme version part, could we do the same?
![Screen Shot 2022-09-20 at 2 46 26 PM](https://user-images.githubusercontent.com/1691518/191187207-5f797e74-518f-4848-a5b1-42b9912fa87f.png)

Reply:
fixed